### ISSUE TITLE
applyToSources gives an error when applyToBranches is correctly specified

### ISSUE THREAD

**Issue by rcgee** [gem/oq-engine#4884]
This error occurs when we have a source model logic tree with more than one branch and we want to consider epistemic uncertainty of a source, and the source belongs to one source model branch but not the other.

Since #4503, we force the user to specify applyToBranches if applyToSources is used. However, we still get an error (even with applyToBranches correctly specified) when the source does not exist in the other source model (i.e. the one not specified by applyToBranches). An example is: [applyToBranches.zip](https://github.com/gem/oq-engine/files/3372031/SEA_applyToBranches.zip)

applyToBranches is used in the SEA model. However the mosaic CI has not been failing because we decided to add a "fake" source (with the same ID and rates near 0) in the source model that does not contain this source, to initially get around this issue.

Also, the risk guys are getting the same error message running SEA with event-based (even with the fake source specified), which is why I originally starting looking into this.  I wonder if the two things are related.

```
2019-07-09T08:29:38.90,INFO,MainProcess/4235,rgee running /home/hazard/rgee/201907_helpcata/SEA/in_small/job.ini [--hc=None]

2019-07-09T08:29:38.93,INFO,MainProcess/4235,Using engine version 3.6.0-git38efbb0b90

2019-07-09T08:29:40.45,WARNING,MainProcess/4235,Using celery@holiday, celery@joplin, celery@makeba, celery@winehouse, 80 cores

2019-07-09T08:29:40.54,INFO,MainProcess/4235,Zipping the input files

2019-07-09T08:29:40.66,INFO,MainProcess/4235,Checksum of the input files: 3574864294

2019-07-09T08:29:40.68,INFO,MainProcess/4235,Reading the risk model if present

2019-07-09T08:29:40.85,INFO,MainProcess/4235,Read 30224 hazard sites

2019-07-09T08:29:40.91,INFO,MainProcess/4235,Validated ssmLT.xml in 0.00 seconds

2019-07-09T08:29:40.93,INFO,MainProcess/4235,Potential number of logic tree paths = 54

2019-07-09T08:29:40.96,INFO,MainProcess/4235,Reading the source model(s) in parallel

2019-07-09T08:29:43.01,INFO,MainProcess/4235,Sent 1.63 KB of data in 4 read_source_models task(s)

2019-07-09T08:29:43.09,INFO,MainProcess/4235,read_source_models  25% [of 4 tasks]

2019-07-09T08:29:43.11,INFO,MainProcess/4235,read_source_models  50% [of 4 tasks]

2019-07-09T08:29:43.91,INFO,MainProcess/4235,read_source_models  75% [of 4 tasks]

2019-07-09T08:29:43.99,INFO,MainProcess/4235,read_source_models 100% [of 4 tasks]

2019-07-09T08:29:44.02,INFO,MainProcess/4235,Received 8.63 KB in 2 seconds, biggest output=3.52 KB

2019-07-09T08:29:44.06,CRITICAL,MainProcess/4235,

2019-07-09T08:29:44.08,CRITICAL,MainProcess/4235,Traceback (most recent call last):
  File "/opt/openquake3/oq-engine/openquake/engine/engine.py", line 339, in run_calc
    close=False, **kw)
  File "/opt/openquake3/oq-engine/openquake/calculators/base.py", line 193, in run
    self.pre_execute()
  File "/opt/openquake3/oq-engine/openquake/calculators/base.py", line 510, in pre_execute
    self.read_inputs()
  File "/opt/openquake3/oq-engine/openquake/calculators/base.py", line 405, in read_inputs
    oq, self.monitor(), srcfilter=self.src_filter)
  File "/opt/openquake3/oq-engine/openquake/commonlib/readinput.py", line 904, in get_composite_source_model
    oqparam, gsim_lt, source_model_lt, monitor, in_memory, srcfilter):
  File "/opt/openquake3/oq-engine/openquake/commonlib/readinput.py", line 804, in get_source_models
    (srcid, source_model_lt.filename))
ValueError: The source 287716burma is not in the source model, please fix applyToSources in /home/hazard/rgee/201907_helpcata/SEA/in_small/ssmLT.xml or the source model

```

**Comment by micheles** [comment]
@rcgee , you should provide a simple test case that exhibit the error (like SEA with all sources removed except the ones which are sufficient to show the problem).

**Comment by rcgee** [comment]
Hi @micheles , a simple test case is provided in the zip file above. It contains only the sources showing the problem. 

I did a test considering the simplified logic trees below, which should be equivalent to each other. I found that Case 1 does not work (i.e. applying uncertainty to a source in ssm2.xml, when branchID=b2), while Case 2 does work (i.e. applying uncertainty to a source in ssm2.xml, when branchID=b1). The first case is the one in the zip file. 

In the meantime, I'll change the SEA logic tree to follow Case 2, since I found that it works in both classical and event-based (@CatalinaYepes)

```
Case 1:
<logicTreeBranch branchID="b1"> <uncertaintyModel> ssm1.xml
<logicTreeBranch branchID="b2"> <uncertaintyModel> ssm2.xml
applyToSources="287716burma" applyToBranches="b2"

Case 2:
<logicTreeBranch branchID="b1"> <uncertaintyModel> ssm2.xml
<logicTreeBranch branchID="b2"> <uncertaintyModel> ssm1.xml
applyToSources="287716burma" applyToBranches="b1"
```
